core-bundle/src/Migration/Version412/AllowedExcludedFieldsMigration.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version412;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class AllowedExcludedFieldsMigration extends AbstractMigration
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->getSchemaManager();
+
+        if (!$schemaManager->tablesExist('tl_user_group')) {
+            return false;
+        }
+
```

Comment:
```suggestion

        $count = $this->connection->fetchOne("SELECT COUNT(*) FROM tl_user_group WHERE alexf LIKE '%tl_files::%'");
        
        if ($count > 0) {
            return false;
        }
```

If any `tl_files::` entry exists we should not run the migration IMO.

Reply:
This check is already present on line 49 and the early return on line 53, isn't it?

Reply:
I don’t think so. `0 === \count($groups)` is only true if every single row in `tl_user_group` has a `tl_files::`. But I think we should also skip the migration if a single row has  `tl_files::` in it because that means we are most likeley not in an update to 4.12.

Reply:
Added in 9730741387614720e5e628f53c96dd94457db637.